### ISSUE TITLE
Get Secret Manager Shows Asterisk

### ISSUE THREAD

**Issue by imcodingideas** [GoogleCloudPlatform/github-actions#166]
### TL;DR
I am trying to deploy to google cloud run, with my environment variables using Get Secret Manager. However; the Docker container is not starting.

**Expected behavior**
I would like to be able use your awesome Get Secret Manager to store our secrets. 

**Observed behavior**
My container is not starting. I have a very simple docker-compose.yaml file where I am pass in a local  env_file:
      .env 

and it starts with **docker-compose up.** with no issues. I think I'm missing something silly. I have 2 days on this. Blocked. 

### Reproduction

**Action YAML**

```yaml
name: Setup, Build, and Deploy Staging to Cloud Run

on:
  pull_request:
    branches:
    - staging

env:
  GCP_PROJECT: sendspark-staging
  GCP_REGION: us-central1
  SERVICE_NAME: api-v2-staging

jobs:
  deploy-staging:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout Code
      uses: actions/checkout@v2

    - name: Cache node modules
      uses: actions/cache@v2
      env:
        cache-name: cache-node-modules
      with:
        path: ~/.npm
        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
        restore-keys: |
          ${{ runner.os }}-build-${{ env.cache-name }}-
          ${{ runner.os }}-build-
          ${{ runner.os }}-

    - name: Setup gcloud
      uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
      with:
        version: '303.0.0'
        service_account_email: github-actions-328@sendspark-staging.iam.gserviceaccount.com
        service_account_key: ${{ secrets.RUN_SA_KEY_STAGING }}
        project_id: ${{ env.GCP_PROJECT }}
        export_default_credentials: true

    - id: secrets
      name: Get Google Secret Manager
      uses: GoogleCloudPlatform/github-actions/get-secretmanager-secrets@master
      with:
        credentials: ${{ secrets.RUN_SA_KEY_STAGING }}
        secrets: |-
          NODE_ENV:61428619041/NODE_ENV
          HOST:61428619041/HOST
          PORT:61428619041/PORT
          MONGO_URL:61428619041/MONGO_URL
          MONGO_DB_NAME:61428619041/MONGO_DB_NAME
          MONGO_ID:61428619041/MONGO_ID
          MONGO_SECRET:61428619041/MONGO_SECRET
          JWT_TOKEN_SECRET:61428619041/JWT_TOKEN_SECRET
          POSTMARK_SECRET:61428619041/POSTMARK_SECRET
          POSTMARK_NOTIFICATION_EMAIL:61428619041/POSTMARK_NOTIFICATION_EMAIL
          DASHBOARD_LINK:61428619041/DASHBOARD_LINK
          THUMBNAIL_HOST:61428619041/THUMBNAIL_HOST
          GCLOUD_ASSET_BUCKET_NAME:61428619041/GCLOUD_ASSET_BUCKET_NAME

    - name: get output
      run: echo '${{steps.secrets.outputs.MONGO_URL}}'

    # Build and push image to Google Container Registry
    - name: Build Staging Container
      run: |-
        docker build -t gcr.io/${{ env.GCP_PROJECT }}/${{ env.SERVICE_NAME }}.sendspark.com . && \
        gcloud builds submit \
          --quiet \
          --tag "gcr.io/${{ env.GCP_PROJECT }}/${{ env.SERVICE_NAME }}.sendspark.com:$GITHUB_SHA"
    
    # Deploy image to Cloud Run
    - name: Deploy
      run: |-
        gcloud run deploy ${{ env.SERVICE_NAME }} \
          --quiet \
          --region ${{ env.GCP_REGION }} \
          --image "gcr.io/${{ env.GCP_PROJECT }}/${{ env.SERVICE_NAME }}.sendspark.com:$GITHUB_SHA" \
          --platform "managed" \
          --allow-unauthenticated
      env:
        NODE_ENV: ${{steps.secrets.outputs.NODE_ENV}}
        HOST: ${{steps.secrets.outputs.HOST}}
        PORT: ${{steps.secrets.outputs.PORT}}
        MONGO_URL: ${{steps.secrets.outputs.MONGO_URL}}
        MONGO_DB_NAME: ${{steps.secrets.outputs.MONGO_DB_NAME}}
        MONGO_ID: ${{steps.secrets.outputs.MONGO_ID}}
        MONGO_SECRET: ${{steps.secrets.outputs.MONGO_SECRET}}
        JWT_TOKEN_SECRET: ${{steps.secrets.outputs.JWT_TOKEN_SECRET}}
        POSTMARK_SECRET: ${{steps.secrets.outputs.POSTMARK_SECRET}}
        POSTMARK_NOTIFICATION_EMAIL: ${{steps.secrets.outputs.POSTMARK_NOTIFICATION_EMAIL}}
        DASHBOARD_LINK: ${{steps.secrets.outputs.DASHBOARD_LINK}}
        THUMBNAIL_HOST: ${{steps.secrets.outputs.THUMBNAIL_HOST}}
        GCLOUD_ASSET_BUCKET_NAME: ${{steps.secrets.outputs.GCLOUD_ASSET_BUCKET_NAME}}
```

**Additional information**
As you can see I am running: get output job. However; I always see: *** 

I don't know if that has anything to do with why the docker is not starting because the real error I see: 

ERROR: (gcloud.run.deploy) Cloud Run error: Container failed to start. Failed to start and then listen on the port defined by the PORT environment variable. Logs for this revision might contain more information.

I don't know if this is a bug because I don't see *** or if Im missing something, somewhere. I do know that my dockerfile works, via Dockerfile

**docker-compose.yaml**

```yaml
version: '3.8'
services:
  app:
    build: .
    working_dir: /app_dev
    entrypoint:
      - npm
    command:
      - run
      - dev
    env_file:
      .env
    ports:
      - 3000:3000
    volumes:
      - .:/app_dev
```

**Comment by averikitsch** [comment]
The Github Actions workflow logs will hide secrets using `***`; therefore it seems like it should be correctly getting the secrets. The issue is your container `Failed to start and then listen on the port defined by the PORT environment variable.` In your `docker-compose.yaml`, you are setting the port, but based on the [container contract](https://cloud.google.com/run/docs/reference/container-contract#port) you can not set the port. Instead your app must listen to the PORT env var provided by the runtime. If you must set a specific port, you can use the [port flag](https://cloud.google.com/sdk/gcloud/reference/run/deploy#--port) for the Cloud Run service.

**Comment by imcodingideas** [comment]
@averikitsch thanks so much for replying. You're great.

What's interesting is if I add all of the environment variables in the cloud run container and redeploy from google console it does run, and start just fine. In `Dockerfile` I added: ARG PORT at the end when I'm doing: 

```yaml
        gcloud run deploy ${{ env.SERVICE_NAME }} \
          --quiet \
          --region ${{ env.GCP_REGION }} \
          --image "gcr.io/${{ env.GCP_PROJECT }}/${{ env.SERVICE_NAME }}.sendspark.com:$GITHUB_SHA" \
          --platform "managed" \
          --allow-unauthenticated \
          --port=8080
```

I still get the same issue.

**Comment by imcodingideas** [comment]
I was able to get it. It's because I was passing the HOST in the env. Thank you!
